lisp/ielm.el

```diff
@@ -605,6 +609,11 @@ Customized bindings may be defined in `ielm-map', which currently contains:
             #'ielm-indirect-setup-hook 'append t)
   (setq comint-indirect-setup-function #'emacs-lisp-mode)
 
+  ;; Input history
+  (setq-local comint-input-ring-file-name ielm--history-file-name)
```

Comment:
Will multiple buffers using comint history automatically append to merge when using a history file?